src/middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { headers } from 'next/headers';
+import { auth } from '@/lib/auth';
+
+export async function middleware(request: NextRequest) {
+	const session = await auth.api.getSession({
```

Comment:
for the middleware I literally copy/pasted this snippet https://www.better-auth.com/docs/integrations/next#middleware

(keep matcher as it is)